solidity/contracts/liquidity-protection/LiquidityProtection.sol

```diff
@@ -1388,8 +1386,15 @@ contract LiquidityProtection is TokenHandler, ContractRegistryClient, Reentrancy
         }
 
         (uint256 hi, uint256 lo) = n > _poolAmount ? (n, _poolAmount) : (_poolAmount, n);
-        (uint256 p, uint256 q) = Math.reducedRatio(hi, d, uint256(-1) / lo);
-        return (p * lo) / q;
+        (uint256 p, uint256 q) = Math.reducedRatio(hi, d, MAX_UINT256 / lo);
+
+        if (q != 0) {
+            return p * lo / q;
+        }
+        if (p == 0) {
```

Comment:
can you explain this change? it seems like a logical change in the function return value

Reply:
Yes:
- The `if (q != 0)` part is obvious (avoid possible division by zero and revert without gas refund)
- The `if (p == 0)` part, now that `q == 0` is used here in order to emulate `0/0 == 1`. In other words, when `p == q`, we can just return `lo`
- The default return statement takes place when `p > 0` and `q == 0`, in which case, we divide the positive number `p * lo` (it is positive because we also know that `lo > 0`) by `0`, hence returning the closest thing to infinity, which is `MAX_UINT256`

Reply:
but in case of any issue, returning `MAX_UINT256` from `protectedAmountPlusFee` is going to lead to a disaster, so why would we fallback to that value?

Reply:
Fixed.

Reply:
Fixed again.